spec/engine/miq_ae_method_service/miq_ae_service_model_base_spec.rb

```diff
@@ -111,10 +111,8 @@
       yaml = svc_service.to_yaml
       service.delete
       model_from_yaml = YAML.safe_load(yaml, [MiqAeMethodService::MiqAeServiceService])
-      expect { model_from_yaml.reload }.to raise_error(
-        NoMethodError,
-        "undefined method `reload' for nil:NilClass"
-      )
```

Comment:
@pkomanek As discussed we still want an error from the `reload` method when the `record_exists?` is false.  Similar to what we see when active_record reloads an object with YAML but the record does not exist in the db.

Here is the ruby script I used to prove this out (and the great writeup that explains why this happens [active_weirdness](https://gist.github.com/NickLaMuro/11b2119819dca4408c5510831c368be5)):
```ruby
# Create new AR object
s = Service.create!(:name => "delete me")
s.reload
s.id
# => 389

# Save it to disk as YAML and destroy the object
File.write('test.yaml', s.to_yaml)
s.destroy

# Load the object from the saved file (you can restart console here to ensure it is not in memory)
s2 = YAML.safe_load(File.read('test.yaml'))
# Save ID as before
s2.id
# => 389

# Reload shows the object is not in the db
s2.reload
# => ActiveRecord::RecordNotFound (Couldn't find Service with 'id'=389)

# But save! and update! seem to work even though the object only existin in memory
s2.save!
#  (0.4ms)  BEGIN
# MiqGroup Load (0.7ms)  SELECT  "miq_groups".* FROM "miq_groups" WHERE "miq_groups"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
# MiqGroup Inst Including Associations (18.8ms - 1rows)
# Tenant Load (1.2ms)  SELECT  "tenants".* FROM "tenants" WHERE "tenants"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
# Tenant Inst Including Associations (19.6ms - 1rows)
# Service Exists (0.6ms)  SELECT  1 AS one FROM "services" WHERE "services"."guid" = $1 AND ("services"."id" != $2) LIMIT $3  [["guid", "308ee870-17cc-49ee-b288-64b2c22b63e1"], ["id", 389], ["LIMIT", 1]]
#  (0.3ms)  COMMIT
# => true

s2.update!(:name => "please delete me")
#  (0.3ms)  BEGIN
# Service Exists (0.7ms)  SELECT  1 AS one FROM "services" WHERE "services"."guid" = $1 AND ("services"."id" != $2) LIMIT $3  [["guid", "308ee870-17cc-49ee-b288-64b2c22b63e1"], ["id", 389], ["LIMIT", 1]]
# SQL (0.5ms)  UPDATE "services" SET "name" = $1, "updated_at" = $2 WHERE "services"."id" = $3  [["name", "please delete me"], ["updated_at", "2019-10-22 21:29:13.009851"], ["id", 389]]
#  (0.2ms)  COMMIT
# => true

# Reload again shows nothing is actually persisted
s2.reload
# ActiveRecord::RecordNotFound (Couldn't find Service with 'id'=389)
```

Reply:
@gmcculloug I modified your scenario for our case:

```
s = Service.create!(:name => "delete me")
s.reload
svc_s = MiqAeMethodService::MiqAeServiceService.find(s.id)
yaml = svc_s.to_yaml
s.destroy
svc_s2 = YAML.safe_load(yaml)
svc_s2.reload
# => ActiveRecord::RecordNotFound (Couldn't find related ActiveRecord object)
svc_s.reaload
# => ActiveRecord::RecordNotFound (Couldn't find Service with 'id'=...)
```